src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,135 @@
 package core.basesyntax.service;
 
+import core.basesyntax.NoValidDataException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_LOGIN_LENGTH_MIN = 6;
+    private static final int VALID_LOGIN_LENGTH_MAX = 20;
+    private static final int VALID_PASSWORD_LENGTH_MIN = 6;
+    private static final int VALID_PASSWORD_LENGTH_MAX = 20;
+    private static final int VALID_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validator(user);
+        return storageDao.add(user);
+    }
+
+    private void validator(User user) {
+        checkIfUserHasNoData(user);
+        checkIfUserAlreadyExist(user);
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+    }
+
+    private void checkIfUserHasNoData(User user) {
+        if (user == null) {
+            throw new NoValidDataException("User can't be added. "
+                    + "Please set all fields for users");
+        }
+    }
+
+    private void checkIfUserAlreadyExist(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new NoValidDataException("User with this login already exist");
+        }
+    }
+
+    private void validateLogin(User user) {
+        checkIfUserHasNotLogin(user);
+        checkCharactersInUsersLogin(user);
+        checkIfUserHasInvalidLoginLength(user);
+    }
+
+    private void validatePassword(User user) {
+        checkIfUserHasNotPassword(user);
+        checkCharactersInUsersPassword(user);
+        checkIfUserHasInvalidPasswordLength(user);
+    }
+
+    private void validateAge(User user) {
+        checkIfUserDoesNotSetAge(user);
+        checkIfUserHasInvalidAge(user);
+    }
+
+    private void checkIfUserHasInvalidLoginLength(User user) {
+        int loginLength = user.getLogin().length();
+        if (loginLength < VALID_LOGIN_LENGTH_MIN || loginLength > VALID_LOGIN_LENGTH_MAX) {
+            throw new NoValidDataException("Incorrect login length. "
+                    + "Login must contain at least " + VALID_LOGIN_LENGTH_MIN
+                    + "  and less then " + VALID_LOGIN_LENGTH_MAX + " characters");
+        }
+    }
+
+    private void checkCharactersInUsersLogin(User user) {
+        String login = user.getLogin();
+        boolean onlyLetters = true;
+        for (int i = 0; i < login.length(); i++) {
+            char ch = login.charAt(i);
+            if (!Character.isLetter(ch)) {
+                onlyLetters = false;
+                break;
+            }
+        }
+        if (!onlyLetters) {
+            throw new NoValidDataException("Login must contain only characters");
+        }
+    }
+
+    private void checkIfUserHasNotLogin(User user) {
```

Comment:
And it actually code duplication, It will be better to create method wich accepts object and parameter name, checks if object is null and throws error for specified field